### ISSUE TITLE
Detox Not Properly Rerunning Tests

### ISSUE THREAD

**Issue by ReshmiCode** [wix/Detox#1589]
**Describe the bug**
Based on https://github.com/wix/Detox/issues/1488 the rerun feature should be part of the test runner. I've added jest-circus to the jest setup which enables test reruns. It reruns correctly for unit tests but not for the end to end tests so I believe it's something on Detox's end.

**To Reproduce**

- [x] I have tested this issue on the latest Detox release and it still reproduces

https://github.com/ReshmiCode/Detox-Test
If you run yarn test you can see the unit test logging multiple times. If you run the detox tests the test is only run once.

**Environment (please complete the following information):**
 - Detox: 14.0.0
 - React Native: 0.59.9
 - Node: 10.15.3

**Comment by rotemmiz** [comment]
Are you basing this assumption on something? Did you debug the test runner to see what scope it gets stuck in? If you are certain that this is related to Detox, please provide trace logs (like explained in the issue template).

I have a hunch this is malconfiguration of jest, you put E2E folder in the ignore path
https://github.com/ReshmiCode/Detox-Test/blob/master/package.json#L29

**Comment by ReshmiCode** [comment]
I put e2e in the ignore path so when I run `yarn test` only the unit tests run. Running detox test with this configuration works (the test does pass when I remove the extra character I added to fail it). I removed that statement and the failing test still doesn't rerun. When a failed unit test reruns, it takes longer and does log the statement multiple times. For the e2e test, it only takes a short amount of time to fail so it's not rerunning.
[detoxLog.txt](https://github.com/wix/Detox/files/3477277/detoxLog.txt)
